src/mesh/parallel/shiftedmetric.cxx

```diff
@@ -109,50 +105,44 @@ void ShiftedMetric::calcYUpDown(Field3D &f) {
   Field3D& yup = f.yup();
   yup.allocate();
 
-  for(int jx=0;jx<mesh.LocalNx;jx++) {
-    for(int jy=mesh.ystart;jy<=mesh.yend;jy++) {
-      shiftZ(&(f(jx,jy+1,0)), yupPhs[jx][jy], &(yup(jx,jy+1,0)));
-    }
+  BOUT_FOR(i, f.getRegion("RGN_NOX")) {
+    shiftZ(&(f(i.x(),i.y()+1,0)), yupPhs[i.x()][i.y()], &(yup(i.x(),i.y()+1,0)));
```

Comment:
Again I think you can use `i.yp()`.

Reply:
@d7919: This loop should be using `getRegion2D`, so can't use `i.yp` for `f`. I've updated to follow the rest of your comments.

Reply:
@johnomotani  it's possible to index a `Field3D` with an `Ind2D`, you just need to specify the `z` index as well. This uses a multiply and add to get the corresponding `Ind3D` as opposed to three divides and a modulus to call `i.x()` and `i.y()`, as such I'd expect the `Ind2D` approach to be more efficient. I'll add these changes to the branch that fixes the OpenMP stuff